store/store.go

```diff
@@ -0,0 +1,32 @@
+package store
+
+import (
+	"fmt"
+)
+
+// NotFoundError is returned when a key is not found in the store.
+type NotFoundError struct {
+	Key string
+}
+
+func (e *NotFoundError) Error() string {
+	return fmt.Sprintf("key \"%s\" not found in store", e.Key)
+}
+
+// Store is a generic key value store. Implementations should be safe for concurrent use.
+// It is basically just a map with a lock. Does not yet implement Set so we don't have to
+// worry about this thing growing without bound.
+type Store[T any] interface {
```

Comment:
OK, so I learned that Golang Maps are not threadsafe :)

Reply:
It's thread safe for readers but not if you have something concurrently writing to the map while other things are trying to read from it.